### DISCUSSION TITLE
Mesh permutations/transformations

### DISCUSSION THREAD

**Discussion by jorgensd** [jorgensd/dolfinx_docs#6] [Ideas]
Do we want to cover mesh permutations (the ordering we use, what one needs to know to use them with functions/assembly/interpolation)?

It might be sensible to cover the perm_vtk, perm_gmsh functions that we have implemented as well (not sure they fit here)

**Comment by IgorBaratta** [comment]
Not sure if it fits in a tutorial, but I guess it's important to document.
Maybe not only for end-user but also for developers.

Also maybe we should document mesh reordering techniques or is it in your mesh creation section?

**Reply by jorgensd** [reply]
This is not really a tutorial right; aiming for more low level for advanced usage (as the old tutorial covers the normal use-cases).

By re-ordering technique do you mean how we do permutations (i.e. @mscroggs paper)?

**Reply by IgorBaratta** [reply]
Right!

I mean the mesh is reordered for locality.

**Reply by jorgensd** [reply]
> Right!
> 
> 
> 
> I mean the mesh is reordered for locality.

Oh, that bit. When you say mesh, i guess you mean the vertices and geometry?

I feel like it is clear with a partitioning introduction that cells dont have the same number.

**Reply by IgorBaratta** [reply]
> Oh, that bit. When you say mesh, i guess you mean the vertices and geometry?

Both, topology and geometry.
